frontend/packages/git-service/src/utils/import-strategy-detector.ts

```diff
@@ -59,10 +61,25 @@ export const detectImportStrategies = async (
   const repositoryStatus = gitService ? await gitService.isRepoReachable() : RepoStatus.Unreachable;
   let detectedFiles: string[] = [];
   let detectedCustomData: string[];
+  let pacFiles: string[] = [];
 
   if (repositoryStatus === RepoStatus.Reachable) {
     try {
-      const { files } = await gitService.getRepoFileList();
+      const { files } = await gitService.getRepoFileList({ includeFolder: true });
+      pacFiles = await detectPacFiles(gitService);
+
+      if (
+        isPacRepository(isRepositoryEnabled, gitService, pacFiles) &&
+        !ImportStrategyList.find((strategy) => strategy.type === ImportStrategy.PAC)
+      ) {
+        ImportStrategyList.push({
```

Comment:
`ImportStrategyList` is a const and should not be changed. Doesn't this mutation show Pac always when it was once detected? Also when leaving the import from git page and open it again?

Reply:
At the bottom of the same file, there is another condition, which checks that if it's not a pac repo, remove the object from the ImportStrategyList. So, the situation you are referring to doesn't happen.

```
  if (!isPacRepository(isRepositoryEnabled, gitService, pacFiles)) {
    detectedStrategies = detectedStrategies.filter(
      (strategy) => strategy.type !== ImportStrategy.PAC,
    );
  }
```

Reply:
Are u suggesting cloning the `ImportStrategyList` object and modifying that instead?